Max/src/main/java/springbootApp/java/controllers/InstrumentoFinancieroController.java

```diff
@@ -0,0 +1,74 @@
+package springbootApp.java.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import springbootApp.java.models.Accion;
+import springbootApp.java.models.InstrumentoDTO;
+import springbootApp.java.models.InstrumentoFinanciero;
+import springbootApp.java.models.Tipo;
+import springbootApp.java.services.InstrumentoFinancieroService;
+
+import java.util.ArrayList;
+
+@RestController
+@RequestMapping("/instrumentoFinanciero")
+public class InstrumentoFinancieroController {
+
+    @Autowired
+    private InstrumentoFinancieroService instrumentoFinancieroService;
+
+    public InstrumentoFinancieroController() {
+        this.instrumentoFinancieroService = new InstrumentoFinancieroService();
+    }
```

Comment:
Acá tenes como 2 cosas redundantes que no se entiende bien, por un lado definis el InstrumentoFinancieroService y le agregas el @ Autowired para que lo busque en el SpringContainer y por otro lado lo crear en el Constructor. Por la inversion de control lo vamos a buscar en el SpringContainer, por lo que no es necesario crearlo en el constructor sino simplemente asignarle al atributo viniendo del parametro:

```suggestion
    @Autowired
    private InstrumentoFinancieroService instrumentoFinancieroService;

    public InstrumentoFinancieroController(InstrumentoFinancieroService instrumentoFinancieroService) {
        this.instrumentoFinancieroService = instrumentoFinancieroService;
    }
```